last-will: `weak` attributes cause problems with Clang

When compiling a (correct) solution of the last-will exercise with Clang (version 15.0.7 on Linux) I get the following error:
```
/vagrant/exercism/cpp/last-will/last_will_test.cpp:30:53: error: attribute declaration must precede definition [-Werror,-Wignored-attributes]
    int assemble_account_number(int) __attribute__((weak));
                                                    ^
/vagrant/exercism/cpp/last-will/last_will.cpp:45:9: note: previous definition is here
    int assemble_account_number(int secret_modifier) {
        ^
/vagrant/exercism/cpp/last-will/last_will_test.cpp:31:40: error: attribute declaration must precede definition [-Werror,-Wignored-attributes]
    int assemble_code() __attribute__((weak));
                                       ^
/vagrant/exercism/cpp/last-will/last_will.cpp:50:9: note: previous definition is here
    int assemble_code() {
        ^
```

For some reason Clang doesn't like this `__attribute__((weak))`.  
I couldn't find a quick solution so I'm opening this issue.

--- a/exercises/concept/last-will/last_will_test.cpp
+++ b/exercises/concept/last-will/last_will_test.cpp
@@ -1,3 +1,9 @@
+// Trick to let the code compile, even if the function has not been implemented:
+namespace estate_executor {
+    int assemble_account_number(int) __attribute__((weak));
+    int assemble_code() __attribute__((weak));
+}
+
 #include "last_will.cpp"
 #ifdef EXERCISM_TEST_SUITE
 #include <catch2/catch.hpp>
@@ -24,12 +30,6 @@ TEST_CASE("Family secrets have not been altered") {
     REQUIRE(garcia::blue::code_fragment() == 923);
 }
 
-
-// Trick to let the code compile, even if the function has not been implemented:
-namespace estate_executor {
-    int assemble_account_number(int) __attribute__((weak));
-    int assemble_code() __attribute__((weak));
-}
 TEST_CASE("Account number assembly function exists in correct namespace") {
     REQUIRE_NOTHROW(estate_executor::assemble_account_number(0));
 }

